backends/dpdk/dpdkHelpers.cpp

```diff
@@ -134,27 +134,65 @@ bool ConvertStatementToDpdk::preorder(const IR::AssignmentStatement *a) {
                 src1Op = src1Member;
             }
         }
+        /* If a constant is present in binary operation and it is not 8-bit
+           aligned and it is signed, then the constant value is sign extended
+           with a width of either 32 bit if its original bit-width is less than
+           32 bit or 64 bit if the bit-width of constant is greater than 32 bit
+        */
+
+        if (!isEightBitAligned(src2Op) && src2Op->is<IR::Constant>()) {
+            if (auto tb = src2Op->type->to<IR::Type_Bits>()) {
+                if (tb != nullptr && tb->isSigned) {
+                    isSignExtended = true;
+                    auto consOrgBitwidth = src2Op->to<IR::Constant>()->type->width_bits();
+                    auto bitWidth = consOrgBitwidth < 32 ? 32 : 64;
+                    auto consValue = src2Op->to<IR::Constant>()->value;
+                    auto mask = 1 << (consOrgBitwidth - 1);
+                    auto msb = (consValue & mask) >> (consOrgBitwidth - 1);
+                    auto orgMask = 0;
+                    if (msb) {
+                        for (auto i = consOrgBitwidth; i < bitWidth; i++)
+                            orgMask |= 1 << i;
+
+                        consValue |= orgMask;
```

Comment:
Please try if below solve the prupose (sign-extension):
int tmp_val = val << (sizeof_int - val_bit_width)
val = tmp_val >> (sizeof_int - val_bit_width)